micro_epsilon_scancontrol_driver/src/driver.cpp

```diff
@@ -25,7 +25,7 @@ namespace scancontrol_driver
         private_nh_.param("partial_profile_data_width", config_.pp_point_data_width, 4);
 
         // Create driver interface object:
-        device_interface = CInterfaceLLT();
+        device_interface = new CInterfaceLLT();
```

Comment:
I'm not entirely sure, but in the old implementation, didn't this use the copy-ctor with a temporary on the right side?

If the copy-ctor was not implemented correctly for `CInterfaceLLT` (or the updated SDK has added fields which aren't covered by the (default) copy-ctor), that could lead to `device_interface` being in an inconsistent state.

Reply:
I agree with @gavanderhoorn. It could be the source of the issues with segfaults.

@dave992 Do you have any reasons not to use a smart pointer?

Reply:
Ah, that sounds like a possible reason indeed. I just had a 'what the hell' moment when this change actually solved the issue.

> Do you have any reasons not to use a smart pointer?

Good point. The only 'reason' for doing it like this is that I wanted to get as close to the working SDK examples as possible (they also do not use smart pointers). I'll update the PR and test it again with the sensor.